From 8b5c8e300cb52ad559882d9e17d06d5d4f283480 Mon Sep 17 00:00:00 2001
From: Samir Ahmic <samir@personal.com>
Date: Wed, 15 Apr 2015 15:53:12 +0200
Subject: [PATCH] HBASE-13456 - Improve HFilePrettyPrinter first hbase:meta
 region processing. Filtering  'oldWALs' and 'recovered.edits' in
 HFile#getStoreFiles

---
 .../main/java/org/apache/hadoop/hbase/io/hfile/HFile.java    | 12 +++++++-----
 1 file changed, 7 insertions(+), 5 deletions(-)

diff --git a/hbase-server/src/main/java/org/apache/hadoop/hbase/io/hfile/HFile.java b/hbase-server/src/main/java/org/apache/hadoop/hbase/io/hfile/HFile.java
index 09233a2..8cb75ec 100644
--- a/hbase-server/src/main/java/org/apache/hadoop/hbase/io/hfile/HFile.java
+++ b/hbase-server/src/main/java/org/apache/hadoop/hbase/io/hfile/HFile.java
@@ -832,7 +832,7 @@ public class HFile {
   }
 
   /**
-   * Returns all files belonging to the given region directory. Could return an
+   * Returns all HFiles belonging to the given region directory. Could return an
    * empty list.
    *
    * @param fs  The file system reference.
@@ -842,18 +842,20 @@ public class HFile {
    */
   static List<Path> getStoreFiles(FileSystem fs, Path regionDir)
       throws IOException {
-    List<Path> res = new ArrayList<Path>();
+    List<Path> regionHFiles = new ArrayList<Path>();
     PathFilter dirFilter = new FSUtils.DirFilter(fs);
     FileStatus[] familyDirs = fs.listStatus(regionDir, dirFilter);
     for(FileStatus dir : familyDirs) {
       FileStatus[] files = fs.listStatus(dir.getPath());
       for (FileStatus file : files) {
-        if (!file.isDirectory()) {
-          res.add(file.getPath());
+        if (!file.isDirectory() && 
+            (!file.getPath().toString().contains(HConstants.HREGION_OLDLOGDIR_NAME)) &&
+            (!file.getPath().toString().contains(HConstants.RECOVERED_EDITS_DIR))) {
+          regionHFiles.add(file.getPath());
         }
       }
     }
-    return res;
+    return regionHFiles;
   }
 
   /**
-- 
2.1.0

